src/AccountingModule.sol

```diff
@@ -0,0 +1,173 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.28;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { AccountingToken } from "./AccountingToken.sol";
+import { IFlexStrategy } from "./FlexStrategy.sol";
+
+interface IAccountingModule {
+    event LowerBoundUpdated(uint16 newValue, uint16 oldValue);
+    event TargetApyUpdated(uint16 newValue, uint16 oldValue);
+    event CooldownSecondsUpdated(uint16 newValue, uint16 oldValue);
+    event SafeUpdated(address newValue, address oldValue);
+
+    error TooEarly();
+    error NotSafeManager();
+    error NotStrategy();
+    error InvariantViolation();
+    error TvlTooLow();
+
+    function BASE_ASSET() external view returns (address);
+    function ACCOUNTING_TOKEN() external view returns (AccountingToken);
+
+    function deposit(uint256 amount) external;
+    function withdraw(uint256 amount) external;
+    function processRewards(uint256 amount) external;
+    function processLosses(uint256 amount) external;
+}
+
+contract AccountingModule is IAccountingModule {
+    using SafeERC20 for IERC20;
+
+    uint256 public constant YEAR = 365.25 days;
+    uint256 public constant DIVISOR = 10_000;
+
+    AccountingToken public immutable ACCOUNTING_TOKEN;
+    address public immutable BASE_ASSET;
+    address public immutable STRATEGY;
+
+    address public safe;
+    uint64 public nextRewardWindow;
+    uint16 public cooldownSeconds = 3600;
+    uint16 public targetApy; // in bips;
+    uint16 public lowerBound; // in bips; % of tvl
+
+    modifier checkAndResetCooldown() {
+        if (block.timestamp < nextRewardWindow) revert TooEarly();
+        nextRewardWindow = (uint64(block.timestamp) + cooldownSeconds);
+        _;
+    }
+
+    modifier onlySafeManager() {
+        if (IFlexStrategy(STRATEGY).isSafeManager(msg.sender) == false) revert NotSafeManager();
+        _;
+    }
+
+    modifier onlyStrategy() {
+        if (msg.sender != STRATEGY) revert NotStrategy();
+        _;
+    }
+
+    constructor(
```

Comment:
this still needs to be converted to an initialize()

To be able to AccessControlUpgradeable and initialize whatever else here